api-stat.go

```diff
@@ -126,22 +124,11 @@ func (c Client) statObject(bucketName, objectName string, reqHeaders RequestHead
 	md5sum := strings.TrimPrefix(resp.Header.Get("ETag"), "\"")
 	md5sum = strings.TrimSuffix(md5sum, "\"")
 
-	// Content-Length is not valid for Google Cloud Storage, do not verify.
-	var size int64 = -1
-	if !s3utils.IsGoogleEndpoint(c.endpointURL) {
-		// Parse content length.
-		size, err = strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 64)
-		if err != nil {
-			return ObjectInfo{}, ErrorResponse{
-				Code:       "InternalError",
-				Message:    "Content-Length is invalid. " + reportIssue,
-				BucketName: bucketName,
-				Key:        objectName,
-				RequestID:  resp.Header.Get("x-amz-request-id"),
-				HostID:     resp.Header.Get("x-amz-id-2"),
-				Region:     resp.Header.Get("x-amz-bucket-region"),
-			}
-		}
+	// Parse content length.
+	size, err := strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 64)
+	if err != nil {
+		// Content-Length is not valid, do not verify.
+		size = -1
```

Comment:
The question "Why are we ignoring ParseInt error" arises in mind, previous code was better. Or you can do:

```
@@ -126,11 +124,12 @@ func (c Client) statObject(bucketName, objectName string, reqHeaders RequestHead
        md5sum := strings.TrimPrefix(resp.Header.Get("ETag"), "\"")
        md5sum = strings.TrimSuffix(md5sum, "\"")
 
-       // Content-Length is not valid for Google Cloud Storage, do not verify.
+       // If server does not send Content-Length then we set size to -1.
        var size int64 = -1
-       if !s3utils.IsGoogleEndpoint(c.endpointURL) {
+       lenStr := resp.Header.Get("Content-Length")
+       if lenStr != "" {
                // Parse content length.
-               size, err = strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 64)
+               size, err = strconv.ParseInt(lenStr, 10, 64)
                if err != nil {
                        return ObjectInfo{}, ErrorResponse{
                                Code:       "InternalError",
```